tools/rebasing.md

```diff
@@ -0,0 +1,95 @@
+# Rebasing commits from upstream
+
+In order to stay up to date with changes to the specification, we rebase changes from the upstream repository into our main branch. This isn't a simple `git merge`, as our repos do not share history (and even if they did would almost never merge cleanly). To make this rebase less painful, there's some tooling that can be run to setup and walk through the rebase commit by commit, taking automated actions to resolve more trivial rebase conflicts.
+
+## Getting started
+
+Some definitions / naming conventions used throughout this doc:
+
+ - pub: a git remote naming convention for this repository (js-temporal/temporal-polyfill). Note this is probably not your git repository's default upstream, as anyone using the Pull Request model must first push their changes to their own fork repository and open a pull request.
+   - `git remote add pub git@github.com:js-temporal/temporal-polyfill.git` to add this remote to your local repository
+ - main: the local branch name for your repo fork's main branch. This should probably track your remote's main branch **not js-temporal/temporal-polyfill's main branch** so you don't accidentally try and push straight to main.
+ - spec-pub: the upstream Temporal proposal repository. This remote is unlikely to be in your local repository, but to perform rebases you will need to fetch it's content into your local git repository.
+   - `git remote add spec-pub https://github.com/tc39/proposal-temporal.git` to add this remote to your local repository.
+
+## Getting the latest changes
+
+Make sure that your local copy of both the `pub` and `spec-pub` repositories are up to date: `git fetch pub && git fetch spec-pub`
+
+Make sure your `main` branch is lined up with the `pub` main branch:
+
+- `git checkout main && git merge --ff-only pub/main`
+
+Initialize the test262 submodule if you haven't before:
+
+ - `git submodule update`
+
+Using the git log, determine the most recent upstream commit that was successfully migrated over using this tooling:
+
+ - `git log --grep 'UPSTREAM_COMMIT=' pup/main`, take note of the commit hash from the description, denoted in this doc as LATEST_UPSTREAMED_COMMIT. Consider exporting this value in your terminal so you can reference it later in commands: `export LATEST_UPSTREAMED_COMMIT=<value>`
+
+Install the rebase-tools dependencies:
+
+ - `cd tools && npm ci`
+
+Check that you can run the tool:
+
+ - `npx ts-node rebase-upstream-commits.ts` should print out the tool's command-line help.
+
+Throughout this document, we refer back to this command as `trt`, and we recommend you add an alias to your current terminal window that can run the tool:
+
+ - `alias trt=$(npx ts-node rebase-upstream-commits.ts realcmd)`
+ - Note this alias refers to live paths, and so as you perform the rebase the content of the rebase tool might no longer be available.
```

Comment:
It might be possible that in some rebase step the current working tree of the repo no longer has any of the rebase tooling available. Because the alias just encodes that exact sequence of path to ts-node and path to the ts script it is likely to stop working in this scenario.

This is actually one scenario I haven't seen myself because I haven't run the tool from within the repo being rebased - I've always housed it somewhere else temporarily and then set the repo during the setup run (hence all the code in this to set a repo flag). I imagine once this is merged I'll have to update either these docs, the tool, or both, to fix it.

If rebasing the repo where the tool lives breaks it, then one possible workaround would be to bundle up ts-node and the script into some temporary bundle that could be stashed somewhere (in /tmp?) for the duration of the rebase. Know of anything like that?

Reply:
Wow, nope sorry. Thanks for the explanation. You may want to think about how to recover manually from this weird case?

Reply:
I'm going to merge the rebase tooling into the repo and test it out, and I'll document any problems I run into and their resolution in a separate PR.